### DISCUSSION TITLE
How is a lunatic process different from v8 isolates?

### DISCUSSION THREAD

**Discussion by jugaadi** [lunatic-solutions/lunatic#35] [General]
Would be really helpful if you can explain in detail. 

Thanks.

**Comment by withtypes** [comment]
We don't know much about V8 implementation details. Can you give us a quick explanation what isolates are in V8?

**Reply by jugaadi** [reply]
This blog gives a better view on v8 isolates than I could ever explain - https://blog.cloudflare.com/cloud-computing-without-containers/

**Reply by gaolaowai** [reply]
Seems similar, except it's running on V8 and closed-source.

It also seems to be very isolated processes, without the option for message passing between them. I doubt it allows supervision trees (a BEAM and *now* Lunatic thing :smile: ), which means the ability to build robust fault-tolerant/resistent applications is limited.

But the idea of microthreads, isolation, WASM (and JS it seems) is all there.

I'm still working and experimenting with Lunatic myself here while learning Rust specifically for this purpose.
